src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,357 @@
-
-
-def create_reference_index_hc(job, shared_ids, input_args):
-    """
-    Uses samtools to create reference index file in working directory,
-    spawns next job in pipeline - create reference dictionary
+    get_bai.addFollowOn(haplotype_caller)
+    return haplotype_caller.rv()
 
-    :param job: Job instance
-    :param shared_ids: dictionary of shared file promises
-    :param input_args: dictionary of input arguments
-    """
-    # Unpack convenience variables for job
-    work_dir = job.fileStore.getLocalTempDir()
-    # Retrieve file path
-    # FIXME: unused variable
-    ref_path = return_input_paths(job, work_dir, shared_ids, 'ref.fa')
-    faidx_output = os.path.join(work_dir, 'ref.fa.fai')
-    # Call: Samtools
-    faidx_command = ['faidx', 'ref.fa']
-    inputs= ref_path
-    outputs={'ref.fa.fai': None}
-    docker_call(work_dir = work_dir,
-                parameters = faidx_command,
-                tool = 'quay.io/ucsc_cgl/samtools',
-                inputs=inputs,
-                outputs=outputs)
-    # Update fileStore for output
-    shared_ids['ref.fa.fai'] = job.fileStore.writeGlobalFile(faidx_output)
-    job.addChildJobFn(create_reference_dict_hc, shared_ids, input_args)
 
-
-def create_reference_dict_hc(job, shared_ids, input_args):
+def download_shared_files(job, config):
     """
-    Uses Picardtools to create sequence dictionary for reference genome.
-    Calls next step in pipeline - spawn batch jobs
+    Downloads reference files shared by all samples in the pipeline
 
-    :param job: Job instance
-    :param shared_ids: dictionary of shared file promises
-    :param input_args: dictionary of input arguments
+    :param config dict: pipeline configuration options
+    :return dict: updated config dictionary with shared fileStoreIDS
     """
-    # Unpack convenience variables for job
-    work_dir = job.fileStore.getLocalTempDir()
-    # Retrieve file path
-    # FIXME: unused variable
-    ref_path = return_input_paths(job, work_dir, shared_ids, 'ref.fa')
-    # Call: picardtools
-    picard_output = os.path.join(work_dir, 'ref.dict')
-    command = ['CreateSequenceDictionary', 'R=ref.fa', 'O=ref.dict']
-    inputs=['ref.fa']
-    outputs={picard_output: None}
-    docker_call(work_dir = work_dir,
-                env={'JAVA_OPTS':'-Xmx%sg' % input_args.memory},
-                parameters = command,
-                tool = 'quay.io/ucsc_cgl/picardtools',
-                inputs=inputs,
-                outputs=outputs)
-    # Update fileStore for output
-    shared_ids['ref.dict'] = job.fileStore.writeGlobalFile(picard_output)
-    job.addChildJobFn(spawn_batch_variant_calling, shared_ids, input_args)
+    job.fileStore.logToMaster('Downloading shared reference files')
+    reference_names = ['genome.fa', 'phase.vcf', 'mills.vcf', 'dbsnp.vcf', 'hapmap.vcf', 'omni.vcf']
+    if config['run_bwa']:
+        bwa_references = ['amb', 'ann', 'bwt', 'pac', 'sa']
```

Comment:
Is this a hg19-only pipeline? If not, should optionally add alt file for alt-aware alignment vs. GRCh38/dedupe this code to adam_gatk_pipeline that already has this supported & tested...